packages/react-data-grid/src/common/cells/headerCells/ResizableHeaderCell.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useRef, cloneElement } from 'react';
+import { CalculatedColumn } from '../../../common/types';
+
+export interface ResizableHeaderCellProps<R> {
+  children: React.ReactElement<React.ComponentProps<'div'>>;
```

Comment:
Small simplification:
```suggestion
  children: React.ReactHTMLElement<HTMLDivElement>;
```
`React.ReactComponentElement<'div'>` is also available, but it won't let us set the ref with it for some reason.

Reply:
Hmm, seems to be failing

> src/HeaderCell.tsx:98:9 - error TS2741: Property 'ref' is missing in type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<any, any, any>)>' but required in type 'ReactHTMLElement<HTMLDivElement>'.

Reply:
Ah, nevermind then.